ambari-server/src/main/java/org/apache/ambari/server/controller/internal/ProvisionClusterRequest.java

```diff
@@ -207,6 +209,16 @@ public ProvisionClusterRequest(Map<String, Object> properties, SecurityConfigura
     }
   }
 
+  // for tests
+  public ProvisionClusterRequest(Blueprint blueprint, Configuration configuration) {
```

Comment:
I'm a little concerned that this constructor does not call one of the other constructors in this class.

That being said, adding a new constructor to facilitate unit tests is fine. 

One minor issue:  Can the access scope of this constructor be limited to package-only access?  For future maintenance, it might be better to keep this out of the public scope (if possible).

Reply:
I would also prefer package-private access, but the test that uses it is in a different package, so that wouldn't work.

Reply:
Ok, that's what I assumed was the reason for making this public, but just wanted to double-check.  

Thanks!